### PR TITLE
Refactor code and Improve readability.

### PR DESCRIPTION
**Pull request by sreeshas**
PR checklist

[* ] Read the contribution guildelines.
[ *] Filed the PR against the correct branch: master for non-breaking changes
Description of the PR

DefaultGenerator.java is very hard to read. The methods are very long and deeply nested. This makes it hard to read and maintain it.

This fix refactors DefaultGenerator by breaking it down into smaller chunks. It also adopts early return principle to avoid deeply nested code.

I ran maven tests and made sure all tests are passing.

### CODE REVIEW

**Comment by sreeshas** [comment]
This change breaks down generate() method in DefaultGenerator into smaller chunks , 

configureGeneratorProperties()
configureSwaggerInfo()
generateApis()
generateModels()
generateModelTests()
generateModelDocumentation()
generateSupportingFiles()
getHost()
getScheme()
buildSupportFileBundle()

This approach makes it easy to read, comprehend, debug and maintain going forward. I have also adopted 'early return' principle to avoid deep nested code.

**Comment by wing328** [comment]
@sreeshas thanks for the PR. I agree your refactoring makes the code more readable.

If no one has any feedback, I'll merge this on coming Wed after reviewing it one more time.

Thank you for the contribution.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java [hunk 1]
```diff
@@ -533,122 +620,33 @@ public int compare(CodegenSecurity one, CodegenSecurity another) {
...
+        InlineModelResolver inlineModelResolver = new InlineModelResolver();
+        inlineModelResolver.flatten(swagger);
+
+        List<File> files = new ArrayList<File>();
+        // models
+        List<Object> allModels = new ArrayList<Object>();
+        generateModels(files, allModels);
+        // apis
+        List<Object> allOperations = new ArrayList<Object>();
+        generateApis(files, allOperations);
```
Comment (sreeshas) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java]: Sounds good. I'm working on this fix.

Comment (sreeshas) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java]: https://github.com/swagger-api/swagger-codegen/pull/4304

**Pull request merged**